components/brave_wallet/browser/eth_tx_state_manager.cc

```diff
@@ -222,114 +100,8 @@ std::unique_ptr<EthTxStateManager::TxMeta> EthTxStateManager::ValueToTxMeta(
-  const base::Value* value =
-      dict->FindPath(GetNetworkId(prefs_, chain_id_) + "." + id);
-  if (!value)
-    return nullptr;
-
-  return ValueToTxMeta(*value);
-}
-
-void EthTxStateManager::DeleteTx(const std::string& id) {
-  DictionaryPrefUpdate update(prefs_, kBraveWalletTransactions);
-  base::Value* dict = update.Get();
-  dict->RemovePath(GetNetworkId(prefs_, chain_id_) + "." + id);
-}
-
-void EthTxStateManager::WipeTxs() {
-  prefs_->ClearPref(kBraveWalletTransactions);
-}
-
-std::vector<std::unique_ptr<EthTxStateManager::TxMeta>>
-EthTxStateManager::GetTransactionsByStatus(
-    absl::optional<mojom::TransactionStatus> status,
-    absl::optional<EthAddress> from) {
-  std::vector<std::unique_ptr<EthTxStateManager::TxMeta>> result;
-  const base::Value* dict = prefs_->GetDictionary(kBraveWalletTransactions);
-  const base::Value* network_dict =
-      dict->FindKey(GetNetworkId(prefs_, chain_id_));
-  if (!network_dict)
-    return result;
-
-  for (const auto it : network_dict->DictItems()) {
-    std::unique_ptr<EthTxStateManager::TxMeta> meta = ValueToTxMeta(it.second);
-    if (!meta) {
-      continue;
-    }
-    if (!status.has_value() || meta->status == *status) {
-      if (from.has_value() && meta->from != *from)
-        continue;
-      result.push_back(std::move(meta));
-    }
-  }
-  return result;
-}
-
-void EthTxStateManager::ChainChangedEvent(const std::string& chain_id) {
-  chain_id_ = chain_id;
-  network_url_ = json_rpc_service_->GetNetworkUrl();
-}
-
-void EthTxStateManager::OnAddEthereumChainRequestCompleted(
-    const std::string& chain_id,
-    const std::string& error) {}
-
-void EthTxStateManager::RetireTxByStatus(mojom::TransactionStatus status,
-                                         size_t max_num) {
-  if (status != mojom::TransactionStatus::Confirmed &&
-      status != mojom::TransactionStatus::Rejected)
-    return;
-  auto tx_metas = GetTransactionsByStatus(status, absl::nullopt);
-  if (tx_metas.size() > max_num) {
-    EthTxStateManager::TxMeta* oldest_meta = nullptr;
-    for (const auto& tx_meta : tx_metas) {
-      if (!oldest_meta) {
-        oldest_meta = tx_meta.get();
-      } else {
-        if (tx_meta->status == mojom::TransactionStatus::Confirmed &&
-            tx_meta->confirmed_time < oldest_meta->confirmed_time) {
-          oldest_meta = tx_meta.get();
-        } else if (tx_meta->status == mojom::TransactionStatus::Rejected &&
-                   tx_meta->created_time < oldest_meta->created_time) {
-          oldest_meta = tx_meta.get();
-        }
-      }
-    }
-    DeleteTx(oldest_meta->id);
-  }
-}
-
-void EthTxStateManager::AddObserver(EthTxStateManager::Observer* observer) {
-  observers_.AddObserver(observer);
-}
-
-void EthTxStateManager::RemoveObserver(EthTxStateManager::Observer* observer) {
-  observers_.RemoveObserver(observer);
+std::string EthTxStateManager::GetTxPrefPathPrefix() {
+  return "ethereum." + GetNetworkId(prefs_, json_rpc_service_->GetChainId());
```

Comment:
This should be a constant like `kEthereumPrefKey` so it is consistent among different prefs
I have a similar `kEthereumNetworkKey` for network prefs in 
https://github.com/brave/brave-core/commit/a2c7b33643ed1bcf422a29ea98b7e17855cfbd0c#diff-c5fc7becf8960e213bdd3b95a68f6d48082a0e7c295d4658bed411d7a9a379bbR650-R652
but I think `kEthereumPrefKey` is better considering many prefs will soon have multi coin support

Reply:
Please also use `base::StrCat` to construct the final prefix instead of `+`

Reply:
Fixed in https://github.com/brave/brave-core/pull/12408/commits/c57a9cac691671a4858c1b4c51df13356f16ae81